### ISSUE TITLE
Problem with UPDATE, LOCK RECORD and LIMIT Together in query

### ISSUE THREAD

**Issue by n0lf** [orientechnologies/orientdb#5817]
According to this site you should be allowed to use both UPDATE, LOCK RECORD and LIMIT in one command
```SQL
update Address set primary = true WHERE  street = "some street" LOCK RECORD LIMIT 1
```

But I get the following error:

> com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #0: Error parsing query: 
> select from Address WHERE  street = "some street" LOCK RECORD LIMIT 1
> Encountered " <LIMIT> "LIMIT "" at line 1, column 114.
> Was expecting one of:
> ...
> 

This works however:
```SQL
update (select from Address WHERE  street = "some street" LIMIT 1) set primary = true LOCK RECORD
```

**Comment by luigidellaquila** [comment]
Hi @n0lf 

I reproduced the problem, I'm working to fix it

Thanks

Luigi

**Comment by luigidellaquila** [comment]
HI @n0lf 

I pushed a fix on branches 2.1.x and develop. The fix will be released next week in 2.1.13

Thanks

Luigi
